src/avatar_action.cpp

```diff
@@ -656,8 +657,9 @@ void avatar_action::swim( map &m, avatar &you, const tripoint &p )
 static float rate_critter( const Creature &c )
 {
     const npc *np = dynamic_cast<const npc *>( &c );
+    const item weapon = np->get_wielded_item();
     if( np != nullptr ) {
-        return np->weapon_value( np->weapon );
+        return np->weapon_value( weapon );
```

Comment:
1. Potential null `np` pointer deference at line 660
2. Is it intended to create a copy of `item weapon` rather than a reference or pointer?